main.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	// +kubebuilder:scaffold:imports
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-vsphere/controllers"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/config"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/record"
+)
+
+const (
+	profilerAddrEnvVar  = "PROFILER_ADDR"
+	syncPeriodEnvVar    = "SYNC_PERIOD"
+	requeuePeriodEnvVar = "REQUEUE_PERIOD"
+)
+
+var (
+	defaultProfilerAddr  = os.Getenv(profilerAddrEnvVar)
+	defaultSyncPeriod    = 10 * time.Minute
+	defaultRequeuePeriod = 20 * time.Second
+
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = infrav1.AddToScheme(scheme)
```

Comment:
Need to add `_ = clusterv1.AddToScheme(scheme)` ? when i didn't add this ,i got error : `no kind is registered for the type v1alpha2.Cluster in scheme`